### DISCUSSION TITLE
Associating outputs with model code versions

### DISCUSSION THREAD

**Discussion by samsrabin** [ESMCI/cime#4320] [Q&A]
Hello,

I've been doing CESM runs with various versions of the code, and a brittle aspect of my workflow is the need to double-check what version of the model code any build directory is associated with before starting a run. I also should be manually making a text file in the archived outputs directory with some info about the latest version, but so far have been too lazy.

In the previous model I worked on, I just aliased `make` to include some commands making such files. These included:

- Build settings
- Latest code version (commit number and description; output of `git log -1`)
- MD5 checksum of executable

Obviously such a technique wouldn't work with CIME. Does anything in CIME save any of that sort of info anywhere? If not (or if so but not everything/everywhere I described), I'd be happy to contribute this.

Thanks in advance!

**Comment by jedwards4b** [comment]
Hi Sam,
There is an xml variable in the env_case.xml file called MODEL_VERSION.   For a more detailed listing you can use
checkout_externals -Sv
